Tests/TSCBasicTests/PathTests.swift

```diff
@@ -1,31 +1,75 @@
 /*
  This source file is part of the Swift.org open source project
 
- Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
  Licensed under Apache License v2.0 with Runtime Library Exception
 
  See http://swift.org/LICENSE.txt for license information
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
-*/
+ */
 
 import Foundation
 import TSCBasic
 import TSCTestSupport
 import XCTest
 
+func generatePath(_ length: Int, absolute: Bool = true, duplicateSeparators: Bool = false) -> String {
+    #if !os(Windows)
+    var path = absolute ? "/" : ""
+    let separator = duplicateSeparators ? "//" : "/"
+    #else
+    var path = absolute ? #"C:"# : ""
+    let separator = duplicateSeparators ? #"\\"# : #"\"#
+    #endif
+    var currentPathLength = path.count
+    var dirNameCount = 0
+    while currentPathLength < length {
+        let dirName = String(dirNameCount)
+        path.append("\(path.count != 0 ? separator : "")\(dirName)")
```

Comment:
Note that path _components_ on Windows generally have a maximum of 255 characters; you might want to account for that here.